### PR TITLE
Fix/6584 content type for lakectl local

### LINKED ISSUE #6584: Currently lakectl local doesn't support upload with content type.
https://github.com/treeverse/lakeFS/issues/6578

Unlike the web browser, when uploading a file using lakectl it doesn't perform content detection. No content type is treated as "application/octet-stream", and this is why clicking on it will download the object.

We can set the content type while uploading the file using lakectl:
```
lakectl fs upload -s ./abc.pdf --content-type "application/pdf" lakefs://myrepo/origin_corpus/abc.pdf
```

But currently lakectl local doesn't support upload with content type.
I wish it can be supported

### PR DESCRIPTION
**Pull request by 0x-chaitu**
Closes #6584 

### Background

Followed code from lakectl fs upload.

### Bug Fix

If this PR is a bug fix, please let us know about:

1. Problem - lakectl local didnt support --content-type flag.
3. Solution - added content-type support

### Testing Details

How were the changes tested?

They were tested by running lakectl local to upload files with content-type flag

### Contact Details

chaitubhojane@gmail.com

### CODE REVIEW

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Hey @0x-chaitu - thank you for your efforts. We value every contribution to this project, it's all about the community :)

Regarding the code, notice that unlike `lakectl fs upload ...`, `lakectl local commit ...` may upload many objects with different content types at once. When the user request for it explicitly using a flag, we should identify the content automatically and not rely on a single given content that cannot fit all types. Other than that, the PR looks very promising.

**Comment by 0x-chaitu** [comment]
Thanks @itaiad200, 
Yes I will add --include-content-type flag with the automation.

File: cmd/lakectl/cmd/local_commit.go [hunk 1]
```diff
@@ -16,8 +16,10 @@ import (
 )
 
 const (
-	localCommitAllowEmptyMessage = "allow-empty-message"
-	localCommitMessageFlagName   = "message"
+	localCommitAllowEmptyMessage    = "allow-empty-message"
+	localCommitMessageFlagName      = "message"
+	localContentTypeFlagName        = "content-type"
+	localIncludeContentTypeFlagName = "include-content-type"
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/local_commit.go:22]:
```suggestion
	localIncludeContentTypeFlagName = "detect-content-type"
```

File: cmd/lakectl/cmd/local_commit.go [hunk 2]
```diff
@@ -120,7 +124,7 @@ var localCommitCmd = &cobra.Command{
 		}()
 		sigCtx := localHandleSyncInterrupt(cmd.Context(), idx, string(commitOperation))
 		s := local.NewSyncManager(sigCtx, client, syncFlags.parallelism, syncFlags.presign)
-		err = s.Sync(idx.LocalPath(), remote, c)
+		err = s.Sync(idx.LocalPath(), remote, c, contentType, includeContentType)
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/local_commit.go:127]: I don't understand.  When would I want to set contentType to the _same_ value on all the files in a Sync?

**Review by arielshaqed — CHANGES_REQUESTED** [review]
I don't understand what the `--content-type` flag does.  AFAICT it gives the _same_ content type to all objects in my commit.  Can you give a use-case for why this would be useful?

**Comment by arielshaqed** [comment]
# THANKS

for picking up this issue and working to fix it.  This is really appreciated, and a great way to improve lakeFS as a product.

**Comment by nopcoder** [comment]
Hi @0x-chaitu, could you please provide an update on the PR status?

**Comment by 0x-chaitu** [comment]
> Hi @0x-chaitu, could you please provide an update on the PR status?

Hi @nopcoder, @itaiad200 has self-assigned the issue.

**Comment by itaiad200** [comment]
Hey @0x-chaitu , I self-assigned the PR so I could track this better. We do that normally for all contributors. Do you still intend to work on this PR or should we take it?

**Comment by 0x-chaitu** [comment]
Hey, @itaiad200 Please take it.

**Pull request closed without merging**